packages/at_client/lib/src/encryption_service/abstract_atkey_encryption.dart

```diff
@@ -201,59 +269,62 @@ abstract class AbstractAtKeyEncryption implements AtKeyEncryption {
   }
 
   /// Stores the encryptedSharedKey for future use.
-  /// Optionally set shouldSync parameter to false to avoid the key to sync to cloud secondary
-  /// Defaulted to sync the key to cloud secondary.
-  void _storeSharedKey(AtKey atKey, String encryptedSharedKey,
-      {bool shouldSync = true}) async {
+  Future<void> _storeMyEncryptedCopyOfSharedSymmetricKey(
+      AtKey atKey, String encryptedSharedKey, Secondary secondary) async {
     var updateSharedKeyForCurrentAtSignBuilder = UpdateVerbBuilder()
       ..atKey =
           '$AT_ENCRYPTION_SHARED_KEY.${atKey.sharedWith?.replaceAll('@', '')}'
       ..sharedBy = atKey.sharedBy
       ..value = encryptedSharedKey;
-    await _atClient
-        .getLocalSecondary()!
-        .executeVerb(updateSharedKeyForCurrentAtSignBuilder, sync: shouldSync);
+    await secondary.executeVerb(updateSharedKeyForCurrentAtSignBuilder,
+        sync: false);
   }
 
   /// Gets the encrypted shared key from the given secondary instance - Local Secondary or Remote Secondary
   ///
   /// Throws [KeyNotFoundException] is key is not found the secondary
-  Future<String?> _getEncryptedSharedKey(
+  Future<String?> _getMyEncryptedCopyOfSharedSymmetricKey(
       Secondary secondary, AtKey atKey) async {
     var llookupVerbBuilder = LLookupVerbBuilder()
       ..atKey =
           '$AT_ENCRYPTION_SHARED_KEY.${atKey.sharedWith?.replaceAll('@', '')}'
       ..sharedBy = atKey.sharedBy;
-    return await secondary.executeVerb(llookupVerbBuilder);
+    String? myCopy;
+
+    try {
+      myCopy = await secondary.executeVerb(llookupVerbBuilder);
+      // ignore: unused_catch_clause, empty_catches
+    } on KeyNotFoundException catch (ignore) {}
+    if (myCopy == 'data:null') {
+      myCopy = null;
+    }
+    if (myCopy != null && myCopy.startsWith('data:')) {
+      myCopy = myCopy.replaceFirst('data:', '');
+    }
+    return myCopy;
   }
 
```

Comment:
The next bit of diff here is kind of rubbish - I removed an obsolete method called isEncryptedSharedKeyInSync. Unrelated, I added a new method called _getTheirEncryptedCopyOfSharedSymmetricKey